src/types/account-id.ts

```diff
@@ -1,45 +1,55 @@
-import { makeClass } from "../utils/make-class";
-const { decodeAccountID, encodeAccountID } = require("ripple-address-codec");
-const { Hash160 } = require("./hash-160");
+import { decodeAccountID, encodeAccountID } from "ripple-address-codec";
+import { Hash160 } from "./hash-160";
 
-const AccountID = makeClass(
-  {
-    AccountID(bytes) {
-      Hash160.call(this, bytes);
-    },
-    inherits: Hash160,
-    statics: {
-      from(value) {
-        return value instanceof this
-          ? value
-          : /^r/.test(value)
-          ? this.fromBase58(value)
-          : new this(value);
-      },
-      cache: {},
-      fromCache(base58) {
-        let cached = this.cache[base58];
-        if (!cached) {
-          cached = this.cache[base58] = this.fromBase58(base58);
-        }
-        return cached;
-      },
-      fromBase58(value) {
-        const acc = new this(decodeAccountID(value));
-        acc._toBase58 = value;
-        return acc;
-      },
-    },
-    toJSON() {
-      return this.toBase58();
-    },
-    cached: {
-      toBase58() {
-        return encodeAccountID(this._bytes);
-      },
-    },
-  },
-  undefined
-);
+/**
+ * Class defining how to encode and decode an AccountID
+ */
+class AccountID extends Hash160 {
```

Comment:
This class is public right? 

I've seen most Javascript use `AccountId` as idiomatic casing. My guess is that re-casing is more effort than it is worth and would cause breaking API changes though.

Reply:
Lots of the default types aren't in CamelCase, like `STObject` or `AccountID`. The names of the classes are defined to exactly match their names in `defintions.json` i.e.

 `"ObjectEndMarker",
      {
        "nth": 1,
        "isVLEncoded": false,
        "isSerialized": true,
        "isSigningField": true,
        "type": "STObject"
      }`

Reply:
Ah, fair enough.